web/src/rest.js

```diff
@@ -5,9 +5,61 @@ import { RestClient } from '@girder/components/src';
 const apiRoot = process.env.VUE_APP_API_ROOT;
 const publishApiRoot = process.env.VUE_APP_PUBLISH_API_ROOT;
 
+// TODO remove the girderId if we can eliminate the girder ID from the URL
+function girderize(publishedDandiset, girderId) {
+  const { // eslint-disable-next-line camelcase
+    created, updated, dandi_id, version, metadata,
+  } = publishedDandiset;
+  return {
+    _id: girderId,
+    created,
+    updated,
+    version,
+    name: dandi_id,
+    lowerName: dandi_id,
+    meta: metadata,
+  };
+}
+
 const girderRest = new RestClient({ apiRoot, setLocalCookie: true });
 const publishRest = axios.create({ baseURL: publishApiRoot });
 
+Object.assign(publishRest, {
```

Comment:
I personally tend to stray away from extending `axios` objects, and towards normal functions like shown in `loggedIn()` and `user()`. Not saying using `Object.assign` is the wrong approach here, but I'm just curious what the benefits are? Is it just for the shorthand?

Reply:
I copied the pattern from the implementation of girder's `RestClient`. I don't really like how decontextualized the raw functions are:
```
import { versions }  from '@/rest';
... lots of code ...
const myVersions = await versions(identifier); // ??? where did this come from? why async?
```
vs.
```
import { publishRest } from `@/rest`;
... lots of code ...
const myVersions = await publishRest.versions(identifier); // ah yes, an API call
```
 Especially now that we have two different client objects, I think it's helpful to have the namespacing provided by having the functions belong to their respective clients.

I admit I'm a little biased because I've used Java a lot and this is how the Java implementation would be set up.
`Object.assign` is gross, but I like the more explicit API surface it provides. Are there benefits to just exporting the raw functions I'm not thinking of?